python/cugraph/structure/graph.py

```diff
@@ -212,7 +215,26 @@ def add_edge_list(self, source, destination, value=None):
             self.from_cudf_edgelist(input_df)
 
     def from_dask_cudf_edgelist(self, input_ddf):
-        self.edgelist = self.EdgeList(input_ddf)
+        if self.edgelist is not None or self.adjlist is not None:
+            raise Exception('Graph already has values')
+        if type(self) is Graph:
+            raise Exception('Undirected distributed graph not supported')
+        if isinstance(input_ddf, dask_cudf.DataFrame):
+            self.distributed = True
+            self.local_data = None
+            self.edgelist = self.EdgeList(input_ddf)
+        else:
+            raise Exception('input should be a dask_cudf dataFrame')
+
+    def compute_local_data(self, by):
```

Comment:
I like the concise design. Missing test and doc though.

Reply:
added both